embeddings/data/datamodule.py

```diff
@@ -0,0 +1,167 @@
+        self.process_data(stage)
+
+    def process_data(self, stage: Optional[str] = None) -> None:
+        if stage == "fit":
+            for split in self.dataset.keys():
+                self.dataset[split] = self.dataset[split].map(
+                    self.convert_to_features,
+                    batched=True,
+                    remove_columns=[self.target_field],
+                )
+                self.columns = [
+                    c for c in self.dataset[split].column_names if c in self.loader_columns
+                ]
+                self.dataset[split].set_format(type="torch", columns=self.columns)
+
+    def train_dataloader(self) -> DataLoader[HuggingFaceDataset]:
+        return DataLoader(self.dataset["train"], batch_size=self.train_batch_size)
+
+    def val_dataloader(self) -> DataLoader[HuggingFaceDataset]:
+        return DataLoader(self.dataset["validation"], batch_size=self.eval_batch_size)
+
+    def test_dataloader(self) -> DataLoader[HuggingFaceDataset]:
+        return DataLoader(self.dataset["test"], batch_size=self.eval_batch_size)
+
+    @abc.abstractmethod
+    def initalize(self) -> None:
+        """Initialize dataset to populate important attributes needed before calling setup."""
+        pass
+
+    @abc.abstractmethod
+    def convert_to_features(
+        self, example_batch: Dict[str, Any], indices: Optional[List[int]] = None
+    ) -> BatchEncoding:
+        pass
+
+
+class TextClassificationDataModule(HuggingFaceDataModule):
+    def __init__(
+        self,
+        model_name_or_path: str,
+        dataset_name: str,
+        text_fields: Union[str, List[str]],
+        target_field: str,
+        max_seq_length: int = 128,
+        train_batch_size: int = 32,
+        eval_batch_size: int = 32,
+        input_dim: int = 768,
+        load_dataset_kwargs: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ):
+        if isinstance(text_fields, str):
+            text_fields = [text_fields]
+        assert 1 <= len(text_fields) <= 2
+        self.text_fields = text_fields
+        super().__init__(
+            model_name_or_path=model_name_or_path,
+            dataset_name=dataset_name,
+            target_field=target_field,
+            max_seq_length=max_seq_length,
+            train_batch_siz=train_batch_size,
+            eval_batch_size=eval_batch_size,
+            input_dim=input_dim,
+            load_dataset_kwargs=load_dataset_kwargs,
+            **kwargs,
+        )
+
+    def initalize(self) -> None:
+        self.dataset = datasets.load_dataset(self.dataset_name, **self.load_dataset_kwargs)
+        self.num_labels = len(set(ex[self.target_field] for ex in self.dataset["train"]))
+        self.initialized = True
+
+    def convert_to_features(
+        self, example_batch: Dict[str, Any], indices: Optional[List[int]] = None
+    ) -> BatchEncoding:
+        """Encodes either single sentence or sentence pairs."""
+        if len(self.text_fields) > 1:
```

Comment:
maybe `if len(self.text_fields) == 1 elif if len(self.text_fields) == 2 else just in case more than 2?`